db/migrate/20200714191914_add_nasa_email_field_to_users.rb

```diff
@@ -0,0 +1,8 @@
+class AddNasaEmailFieldToUsers < ActiveRecord::Migration
+  disable_ddl_transaction!
+
+  def change
+    add_column :users, :nasa_email_communication, :boolea, default: false, algorithm: :concurrently
```

Comment:
Don't add a column with a default, it'll lock the table while adding this, https://www.braintreepayments.com/blog/safe-operations-for-high-volume-postgresql/

Reply:
Y'know, I had this same thought, and then I checked the migrations folder and saw [a couple](https://github.com/zooniverse/panoptes/blob/master/db/migrate/20190624094308_add_subjects_counter_cache_to_workflow.rb#L3) recent [migrations](https://github.com/zooniverse/panoptes/blob/master/db/migrate/20190507103007_add_education_attr_to_workflows.rb#L3) using `default:`, and not using DDL transactions, and figured I was okay. Guess not!

Reply:
it depends on the table size. If it's big and busy it'll block which can kill the app depending on the query patterns. users is big and busy so it'll cause errors. 

new tables with no data this is ok, even smallish tables with infrequent access is ok (good to get a baseline from a snapshot) but running these migrations carries downtime risk.

Reply:
I guess workflows isn't huge, but it sure is frequently accessed.